src/main/java/org/concordion/internal/command/VerifyRowsCommand.java

```diff
@@ -48,43 +39,41 @@ public void verify(CommandCall commandCall, Evaluator evaluator, ResultRecorder
         Check.isTrue(obj instanceof Iterable, obj.getClass().getCanonicalName() + " is not Iterable");
         Check.isTrue(!(obj instanceof HashSet) || (obj instanceof LinkedHashSet), obj.getClass().getCanonicalName() + " does not have a predictable iteration order");
         Iterable<Object> iterable = (Iterable<Object>) obj;
-        
-        TableSupport tableSupport = new TableSupport(commandCall);
-        Row[] detailRows = tableSupport.getDetailRows();
 
-        announceExpressionEvaluated(commandCall.getElement());
-        
-        int index = 0;
-        for (Object loopVar : iterable) {
-            evaluator.setVariable(loopVariableName, loopVar);
-            Row detailRow;
-            if (detailRows.length > index) {
-                detailRow = detailRows[index];
-            } else {
-                detailRow = tableSupport.addDetailRow();
-                announceSurplusRow(detailRow.getElement());
-            }
-            tableSupport.copyCommandCallsTo(detailRow);
-            commandCall.getChildren().verify(evaluator, resultRecorder);
-            index++;
-        }
-        
-        for (; index < detailRows.length; index++) {
-            Row detailRow = detailRows[index];
-            resultRecorder.record(Result.FAILURE);
-            announceMissingRow(detailRow.getElement());
+        VerifyRowsStrategy verifyRowsStrategy;
+
+        try {
+            verifyRowsStrategy = detectStrategyClass(commandCall)
+                    .getConstructor(CommandCall.class, Evaluator.class, ResultRecorder.class, Announcer.class, String.class, Iterable.class)
+                    .newInstance(commandCall, evaluator, resultRecorder, listeners, loopVariableName, iterable);
+        } catch (Exception e) {
+            throw new RuntimeException("Verification strategy must declare constructor with arguments: " +
+                    "CommandCall commandCall, Evaluator evaluator, ResultRecorder resultRecorder,\n" +
+                    "Announcer<VerifyRowsListener> listeners, String loopVariableName, Iterable<Object> actualRows");
         }
+
+        verifyRowsStrategy.verify();
     }
-    
-    private void announceExpressionEvaluated(Element element) {
-        listeners.announce().expressionEvaluated(new ExpressionEvaluatedEvent(element));
-    }
 
-    private void announceMissingRow(Element element) {
-        listeners.announce().missingRow(new MissingRowEvent(element));
+    private static final String DEFAULT_STRATEGIES_PACKAGE = "org.concordion.internal.command.strategies.";

```

Comment:
Let's use:

```    private static final String DEFAULT_STRATEGIES_PACKAGE = Default.class.getPackage().getName() + ".";```

instead, in case we ever rename the strategies package.